File: Java/388_version3.java
class Solution {
    public int lengthLongestPath(String input) {
        String[] files = input.split("\n");
        int maxLen = 0,currLen = 0;
        Stack<Integer> stack = new Stack<>();
        for(String f:files){
            int level = f.lastIndexOf("\t")+1;
            int fileLen = f.length()-level+1;
            while(stack.size() > level)
                currLen-=stack.pop();
            currLen+=fileLen;
            if(f.contains("."))
                maxLen = maxLen>currLen-1?maxLen:currLen-1;
            stack.push(fileLen);
        }
        return maxLen;
    }
}

File: 388/388_version3.py
class Solution:
    def lengthLongestPath(self, input: str) -> int:
        files = input.split("\n")
        stack = []
        max_len,curr_len = 0,0 #initialize the length
        for f in files:
            level = f.count("\t")
            file_len = len(f)-level+1
            while len(stack) > level:
                curr_len-=stack.pop()
            curr_len+=file_len
            if "." in f:
                max_len = max([max_len,curr_len-1])
            stack.append(file_len)
        return max_len
